cmd/jaeger/internal/receivers/storagereceiver/factory.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package storagereceiver
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+)
+
+// componentType is the name of this extension in configuration.
+const componentType = component.Type("jaeger_storage_receiver")
+
+func NewFactory() receiver.Factory {
```

Comment:
Sorry, this is confusing to me. It looks like you're trying to make it similar to `exporters/storageexporter`, but I don't understand why. The latter is needed to support a transition from OTEL pipeline into Jaeger storage world, where there could be any number of storage backends. Similar functionality is not needed in OTEL itself because their exporters are one-way (write), while Jaeger Storage is read/write.

However, on the receiver side there is no such situation. If we need to receive over Kafka, we can use official OTEL Kafka receiver directly in the pipeline, without any additional machinery from Jaeger.

Please elaborate.

Reply:
Actually, the `storagereceiver` is only used for integration test. The pipeline I used for the test is `collector -> kafka -> ingester -> remote storage` and the validator checks if the spans stored in remote storage match exactly with the provided data. Therefore, I need `storagereceiver` to retrieve spans from the remote storage for validation.

Here's the diagram of the integration test I created. The first test case verifies if the spans sent to Kafka are correct, and the second one checks the spans stored in the remote storage.
![Untitled-2023-12-16-1017](https://github.com/jaegertracing/jaeger/assets/46216691/189c4e59-095e-483c-a564-9fd3c1ff0ef3)

Reply:
In this case could we move it under integration tests? I see that it is not included in the Components() output, so why does it need to be in jaeger/internal? The reason I don't like it is it looks like another production component. but it is not. 

Also, please add READMEs to new components, and save these awesome diagrams in them!

Reply:
Absolutely! I've moved it under the integration test and added the README.